### PR TITLE
[Visuals] Allow specifying treemacs-indentation in pixels

### PR DESCRIPTION
**Pull request by Jasu**
* `treemacs-indentation` may be specified as `'(10 px)`, making indentation in GUI Emacs a single space 10 pixels wide.
* Indentation strings are cached in a vector, since `propertize` is slow.
* In emacs -nw, if `treemacs-indentation` is specified in pixels, default to two spaces.

**EDIT:** Rationale for this change is that variable-width fonts often have very short spaces, but specifying a high number for indentation would make indentation huge in `emacs -nw`.

### CODE REVIEW

**Comment by Alexander-Miller** [comment]
Could the same api be used to pad file icons' width? If so a theme (when support for it is dropped) could then use dir icons with ▸ & ▾ characters to indicate open & closed state (like for example intellij's icons) and the file icons could be left-padded so things still align.

**Comment by Jasu** [comment]
With some changes, yes. E.g.
```
|--INDENT--|--ICON--|--ALIGNMENT--|--BUTTON-LABEL--|
```
where `ALIGNMENT` would be a single space with ``(propertize " " 'display `(:align-to (,(+ indent-width icon-width))))``

This implementation would be nice since it would not require knowing icon width. I'm using `all-the-icons` for my extensions, and getting the glyph width is very hard.

File: src/elisp/treemacs-customization.el [hunk 1]
```diff
@@ -68,9 +69,17 @@
...
-  :type 'integer
+  "The number of spaces or pixels each level is indented in the file tree.
+If the value is integer, indentation is created by repeating
+`treemacs-indentation-string'. If the value is a list of form '(INTEGER px),
+indentation will be a space INTEGER pixels wide."
+  :type '(choice (integer :tag "Spaces" :value 2)
+                 (list :tag "Pixels"
+                       (integer :tag "Pixels" :value 16)
+                       (const :tag "" px)))
+  :set (lambda (key value)
```
[the remark this answers is not in the document]
Comment (Alexander-Miller) [reply src/elisp/treemacs-customization.el]: And if the value is local we could prefill it for the first 6 levels or so.

**Comment by Alexander-Miller** [comment]
Bump?

**Comment by Jasu** [comment]
I'll take a look today, sorry for the delay!

**Comment by Jasu** [comment]
I force-pushed an update that does away with the `:set` functions and insteads keeps a cache key for the indentation variables and checks that instead. I also made the cache and its key buffer-local and initialized it up to 6 levels in `treemacs-mode?

**Comment by Jasu** [comment]
Icons are not being touched here - is there a theme branch or something where it's done (or where I could take a look to implement it?)

**Comment by Alexander-Miller** [comment]
I've pushed my local master, including the theme change, to the `themes` branch. You'll want to look at the icons module, and specifically the internals of the `treemacs-create-icon` macro.

**Comment by Alexander-Miller** [comment]
:+1:

**Pull request merged**